### PR TITLE
[ADD] account_receipt_report: new module to print payment receipt

### PR DESCRIPTION
**Pull request by cubells**
Now you can print a payment receipt and validate on register an invoice payment.

cc @Tecnativa

### CODE REVIEW

File: account_receipt_report/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Account Receipt Report',
+    'version': '11.0.1.0.0',
+    'category': 'Generic Modules/Payment',
+    'author': 'Odoo Community Association (OCA)',
+    'summary': 'Add option to print account payment receipt',
+    'website': 'https://github.com/OCA/account-payment',
+    'license': 'LGPL-3',
```
Comment (pedrobaeza) [inline account_receipt_report/__manifest__.py]: AGPL-3

File: account_receipt_report/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2019 Tecnativa <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Account Receipt Report',
+    'version': '11.0.1.0.0',
+    'category': 'Generic Modules/Payment',
```
Comment (pedrobaeza) [inline account_receipt_report/__manifest__.py]: Generic Modules?

File: account_receipt_report/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    'author': 'Odoo Community Association (OCA)',
```
Comment (pedrobaeza) [inline account_receipt_report/__manifest__.py]: And Tecnativa?

File: account_receipt_report/README.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+   :local:
+
+Usage
+=====
+
+#. Go to any invoice and register a payment.
+#. In payment wizard you will see a new button **Validate and print receipt**.
+   If you click in this button, a receipt of the payment will be printed and
+   the invoice will be payed.
+#. Or you can go to a bank statement and you can print the report when line
```
Comment (pedrobaeza) [inline account_receipt_report/README.rst]: The receipt is printed from bank statement line, but not when going to be reconciled.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_receipt_report/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#. Go to any invoice and register a payment.
+#. In payment wizard you will see a new button **Validate and print receipt**.
+   If you click in this button, a receipt of the payment will be printed and
+   the invoice will be payed.
```
Comment (yajo) [inline account_receipt_report/readme/USAGE.rst:4]: "paid"

File: account_receipt_report/models/account_bank_statement.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+class AccountBankStatementLine(models.Model):
+    _inherit = 'account.bank.statement.line'
+
+    def action_print_invoice_payment(self):
+        # Print the report based in account.move.lines
+        receipt_report = self.env.ref(
+            'account_receipt_report.action_account_report_payment_receipt')
+        account_move_lines = self.journal_entry_ids.filtered(
+            lambda x: x.account_id.internal_type in ['receivable', 'payable']
```
Comment (yajo) [inline account_receipt_report/models/account_bank_statement.py:15]: Use `{'receivable', 'payable'}` instead for performance.

File: account_receipt_report/models/account_payment.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+class AccountPayment(models.Model):
+    _inherit = 'account.payment'
+
+    def action_print_invoice_payment(self):
+        # Print the report
+        receipt_report = self.env.ref(
+            'account_receipt_report.action_account_report_payment_receipt')
+        account_move_lines = self.move_line_ids.filtered(
+            lambda x: x.account_id.internal_type in ['receivable', 'payable']
```
Comment (yajo) [inline account_receipt_report/models/account_payment.py:15]: Same here.

File: account_receipt_report/reports/payment_receipt_report.xml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+                        <style type="text/css">
+                            .square {
+                            border:1px solid;
+                            min-height: 60px;
+                            }
+                            .highlight {
+                            font-weight: bold;
+                            }
+                            .row-content {
+                            padding: 5px;
```
Comment (yajo) [inline account_receipt_report/reports/payment_receipt_report.xml:19]: Would you mind indenting these blocks properly please?

Also, afaik, `<style>` tags are only respected when found in `<head>`. You have to move it there possibly.

**Review by yajo — CHANGES_REQUESTED** [review]
Code OK, just some minor details, although one of them I think it is required.

File: account_receipt_report/reports/payment_receipt_report.xml (hunk 8 above)
Comment (etobella) [reply account_receipt_report/reports/payment_receipt_report.xml:19]: Wouldn't it be easier to create a less file and reference it on the report (also, a class for the report should be necessary)

**Comment by rafaelbn** [comment]
@pedrobaeza please could you review this PR? Thanks!

**Comment by rafaelbn** [comment]
Should we close this one @pedrobaeza ?

**Comment by pedrobaeza** [comment]
Closing for now. Those that want to have it, can rescue it.

**Pull request closed without merging**